src/main/scala/uk/gov/nationalarchives/omega/api/services/Stateful.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2023 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.api.services
+
+import cats.effect.IO
+import org.typelevel.log4cats.{ LoggerFactory, SelfAwareStructuredLogger }
+import org.typelevel.log4cats.slf4j.Slf4jFactory
+import uk.gov.nationalarchives.omega.api.services.ServiceState.Stopped
+
+import java.util.concurrent.atomic.AtomicReference
+
+trait Stateful {
+
+  implicit val loggerFactory: LoggerFactory[IO] = Slf4jFactory[IO]
+  implicit val logger: SelfAwareStructuredLogger[IO] = LoggerFactory[IO].getLogger
+
+  private val state = new AtomicReference[ServiceState](Stopped)
+
+  val invalidState = 99
+
+  def switchState(from: ServiceState, to: ServiceState): IO[Boolean] = IO {
+    val switched = state.compareAndSet(from, to)
+    if (!switched) {
+      logger.error(s"Unable to switch ApiService state from: $from, to: $to.")
```

Comment:
There's a reference to `ApiService` here. What else is going to be `Stateful`?

Reply:
Probably nothing else - but I've removed "ApiService" and it now gets the class name.